packages/flutter_tools/test/integration.shard/generated_plugin_registrant_test.dart

```diff
@@ -0,0 +1,204 @@
+    await _createProject(tempDir.childDirectory('test_plugin'), <String>[
+      '--template=plugin',
+      '--platforms=web',
+      '--project-name',
+      'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+    ]);
+    // The line for the test web plugin (`  TestWebPluginWithAPurposefullyExtremelyLongPackageNameWeb.registerWith(registrar);`)
+    // exceeds 80 chars.
+    // With the above lint rule added, we want to ensure that the `generated_plugin_registrant.dart`
+    // file does not fail analysis (this is a regression test - an ignore was
+    // added to cover this case).
+    await _addDependency(
+        projectDir,
+        'test_web_plugin_with_a_purposefully_extremely_long_package_name',
+        '\n        path: ../test_plugin');
+    await _analyzeProject(projectDir);
+
+    expect(
+        projectDir.childFile('lib/generated_plugin_registrant.dart'), exists);
+  }, overrides: <Type, Generator>{
+    Pub: () => Pub(
+          fileSystem: globals.fs,
+          logger: globals.logger,
+          processManager: globals.processManager,
+          usage: globals.flutterUsage,
+          botDetector: globals.botDetector,
+          platform: globals.platform,
+        ),
+  });
+}
+
+Future<void> _ensureFlutterToolsSnapshot() async {
+  final String flutterToolsPath = globals.fs.path.absolute(globals.fs.path.join(
+    'bin',
+    'flutter_tools.dart',
+  ));
+  final String flutterToolsSnapshotPath =
+      globals.fs.path.absolute(globals.fs.path.join(
+    '..',
+    '..',
+    'bin',
+    'cache',
+    'flutter_tools.snapshot',
+  ));
+  final String dotPackages = globals.fs.path.absolute(globals.fs.path.join(
+    '.packages',
+  ));
+
+  final File snapshotFile = globals.fs.file(flutterToolsSnapshotPath);
+  if (snapshotFile.existsSync()) {
+    snapshotFile.renameSync(flutterToolsSnapshotPath + '.bak');
+  }
+
+  final List<String> snapshotArgs = <String>[
+    '--snapshot=$flutterToolsSnapshotPath',
+    '--packages=$dotPackages',
+    flutterToolsPath,
+  ];
+  final ProcessResult snapshotResult = await Process.run(
+    '../../bin/cache/dart-sdk/bin/dart',
+    snapshotArgs,
+  );
+  if (snapshotResult.exitCode != 0) {
+    print(snapshotResult.stdout);
+    print(snapshotResult.stderr);
+  }
+  expect(snapshotResult.exitCode, 0);
+}
+
+Future<void> _createProject(Directory dir, List<String> createArgs) async {
+  Cache.flutterRoot = '../..';
+  final CreateCommand command = CreateCommand();
+  final CommandRunner<void> runner = createTestCommandRunner(command);
+  await runner.run(<String>[
+    'create',
+    ...createArgs,
+    dir.path,
+  ]);
+}
+
+Future<void> _addDependency(
+  Directory projectDir,
+  String package,
+  String version,
+) async {
+  final File pubspecYaml = projectDir.childFile('pubspec.yaml');
+  expect(pubspecYaml, exists);
+
+  final List<String> lines = await pubspecYaml.readAsLines();
+
+  for (int i = 0; i < lines.length; i++) {
+    final String line = lines[i];
+
```

Comment:
really trivial nit: lots of blank lines here that aren't necessary

(we use blank lines to separate significant parts of the code)